sleap/io/dataset.py

```diff
@@ -629,74 +629,137 @@ def __contains__(self, item) -> bool:
+
+            # Do any conversions first.
+            if isinstance(key, slice):
+                start, stop, step = key.indices(len(self))
+                key = range(start, stop, step)
+            elif isinstance(key, (np.integer, np.ndarray)):
+                key = key.tolist()
+
+            if isinstance(key, int):
+                return self.labels.__getitem__(key)
+
+            elif isinstance(key, Video):
+                if key not in self.videos:
+                    raise KeyError("Video not found in labels.")
+                return self.find(video=key)
+
+            elif isinstance(key, tuple) and len(key) == 2 and isinstance(key[0], Video):
+                if key[0] not in self.videos:
+                    raise KeyError("Video not found in labels.")
+
+                # Do any conversions first.
+                if isinstance(key[1], (np.integer, np.ndarray)):
+                    key = (key[0], key[1].tolist())
+
+                if isinstance(key[1], int):
+                    _hit = self.find_first(
+                        video=key[0], frame_idx=key[1], use_cache=use_cache
                     )
-                return _hit
-            elif isinstance(key[1], (np.integer, np.ndarray)):
-                return self.__getitem__((key[0], key[1].tolist()))
-            elif isinstance(key[1], (list, range)):
-                return self.find(video=key[0], frame_idx=key[1])
-            else:
-                raise KeyError("Invalid label indexing arguments.")
-
-        elif isinstance(key, slice):
-            start, stop, step = key.indices(len(self))
-            return self.__getitem__(range(start, stop, step))
-
-        elif isinstance(key, (list, range)):
-            return [self.__getitem__(i) for i in key]
+                    if _hit is None:
+                        raise KeyError(
+                            f"No label found for specified video at frame {key[1]}."
+                        )
+                    return _hit
+                elif isinstance(key[1], (list, range)):
+                    return self.find(video=key[0], frame_idx=key[1])
+                else:
+                    raise KeyError("Invalid label indexing arguments.")
 
-        elif isinstance(key, (np.integer, np.ndarray)):
-            return self.__getitem__(key.tolist())
+            elif isinstance(key, (list, range)):
+                return [self.__getitem__(i) for i in key]
 
-        else:
-            raise KeyError("Invalid label indexing arguments.")
-
-    def get(self, *args) -> Union[LabeledFrame, List[LabeledFrame]]:
-        """Get an item from the labels or return `None` if not found.
+            else:
+                raise KeyError("Invalid label indexing arguments.")
 
-        This is a safe version of `labels[...]` that will not raise an exception if the
-        item is not found.
-        """
-        try:
-            return self.__getitem__(*args)
         except KeyError:
             return None
```

Comment:
Maybe I'm misreading this but it looks like there's no way to actually have those errors raised, even when calling from `__getitem__` since it'll get caught here and return `None`.

Do we maybe want to have a kwarg in this method that toggles re-raising the exceptions when getting to this line? Something like:

```
def get(self, key, *secondary_key, use_cache=False, raise_error=False):
    try:
        ...
    except KeyError as e:
        if raise_error:
            raise e
        else:
            return None
```


Then we can call it from `__getitem__` with:
```
self.get(key, *secondary_key, raise_error=True)
```

Reply:
You are correct, any errors raised now just return `None`. I was unsure if we actually wanted errors raised or if we preferred returning `None`. I can add a `kwarg` for raising errors.